### ISSUE TITLE
Suggestions for documentation page

### ISSUE THREAD

**Issue by chrisnojima** [facebook/immutable-js#1182]
Hey,

Thanks for the great library!

Ran into two confusing aspects of the docs.

1. The linking seems broken. I didn't dive into the site to see if you're doing something tricky but the anchor jumping seems broken to me (in chrome at least). For example if i
- Start at the root of docs (http://facebook.github.io/immutable-js/docs/#/)
- Click Map on the right navigation panel (http://facebook.github.io/immutable-js/docs/#/Map)
- Click Delete on the right navigation panel (http://facebook.github.io/immutable-js/docs/#/Map/delete)
- Click 'withMutations' link at the bottom of that paragraph (http://facebook.github.io/immutable-js/docs/#/Map/withMutations) 

it sends me to the top of the Map section and doesn't go into withMutations itself.

2. The docs aren't versioned in any way. I haven't jumped into the 4.x-rcs yet (assuming its not really mainline yet) so i started using deleteAll() and was trying to figure out why the method didn't exist yet.

Thanks for reading!

**Comment by bdwain** [comment]
Agreed. It would be super helpful to be able to refer to 3.x docs for people not using the 4.x RC yet. 

Thanks!

**Comment by leebyron** [comment]
Added version to the docs.

We have a couple issues tracking some of the broken links - many of which have been fixed.

Thanks for your comments!

**Comment by hnrysmth** [comment]
Hey there! Quick addition to this issue 😄 

I just spent maybe 20 minutes, half an hour digging into this `deleteAll is not a function` error until I found this [Stack Overflow answer](https://stackoverflow.com/a/43054563):

> Looks like it's only been added on the latest version (ver. 4 RC1 & RC2). I looked over the src code and it's not found in the 3.8.1. Unless you want to use 4.0.0-rc.2, there's no other way to use that method yet.

I guess the most ideal thing would be if each function in the docs had an annotation saying the version number when it was added. Cos I was on the latest stable version so I figured it just meant I could use everything here:

![screen shot 2017-10-09 at 22 31 18](https://user-images.githubusercontent.com/566159/31357329-9847ee86-ad41-11e7-941f-b804e48f1e93.png)

Certainly not saying this is an urgent problem or anything and it's awesome that the docs now say what version they refer to. But if anyone's reading this who particularly wants to make a documentation contribution to this project, I'd say this idea might be a pretty cool candidate to improve it even more! ❤️
